src/main/kotlin/senior/joinu/candid/IDLType.kt

```diff
@@ -235,150 +249,147 @@ sealed class IDLType {
+            override var text = "int8"
         }
 
         object Int16 : Primitive() {
-            override fun toString() = "Int16"
-
-            override fun poetize() = CodeBlock.of("%T", Int16::class).toString()
+            override var text = "int16"
         }
 
         object Int32 : Primitive() {
-            override fun toString() = "Int32"
-
-            override fun poetize() = CodeBlock.of("%T", Int32::class).toString()
+            override var text = "int32"
         }
 
         object Int64 : Primitive() {
-            override fun toString() = "Int64"
-
-            override fun poetize() = CodeBlock.of("%T", Int64::class).toString()
+            override var text = "int64"
         }
 
         object Float32 : Primitive() {
-            override fun toString() = "Float32"
-
-            override fun poetize() = CodeBlock.of("%T", Float32::class).toString()
+            override var text = "float32"
         }
 
         object Float64 : Primitive() {
-            override fun toString() = "Float64"
-
-            override fun poetize() = CodeBlock.of("%T", Float64::class).toString()
+            override var text = "float64"
         }
 
         object Bool : Primitive() {
-            override fun toString() = "Bool"
-
-            override fun poetize() = CodeBlock.of("%T", Bool::class).toString()
+            override var text = "bool"
         }
 
         object Text : Primitive() {
-            override fun toString() = "Text"
-
-            override fun poetize() = CodeBlock.of("%T", Text::class).toString()
+            override var text = "text"
         }
 
         object Null : Primitive() {
-            override fun toString() = "Null"
-
-            override fun poetize() = CodeBlock.of("%T", Null::class).toString()
+            override var text = "null"
         }
 
         object Reserved : Primitive() {
-            override fun toString() = "Reserved"
-
-            override fun poetize() = CodeBlock.of("%T", Reserved::class).toString()
+            override var text = "reserved"
         }
 
         object Empty : Primitive() {
-            override fun toString() = "Empty"
-
-            override fun poetize() = CodeBlock.of("%T", Empty::class).toString()
+            override var text = "empty"
         }
     }
 
     sealed class Other : IDLType() {
         data class Custom(val opcode: Int) : Other() {
+            val text: String = TODO("Not yet implemented")
+            override fun toString() = TODO("Not yet implemented")
             override fun poetize() = CodeBlock.of("%T($opcode)", Custom::class).toString()
         }
 
         data class Future(val opcode: Int) : Other() {
+            val text: String = TODO("Not yet implemented")
+            override fun toString() = TODO("Not yet implemented")
             override fun poetize() = CodeBlock.of("%T($opcode)", Future::class).toString()
         }
     }
+
```

Comment:
Please, consider moving this function in the `Utils.kt` file.

Also, in Kotlin you can use """text""" (3 pairs of double-quotes) to produce multiline strings (all whitespace characters are preserved inside these).

Reply:
And could you also make it public?

Reply:
Did that; however moving the function under Kotlin package `transpile` seems a little misleading; also I couldn't find a more readable representation using the triple-double-quote notation as one has to iterate over the items and indent and append a new line character to each item; please improve upon as you like.